packages/commonwealth/server/workers/evmChainEvents/nodeProcessing.ts

```diff
@@ -72,6 +97,60 @@ export async function processChainNode(
         await lastProcessedBlock.save({ transaction });
       }
 
+      const contestEvents = events
+        .map((event) => {
+          const timestamp = new Date(); // TODO: get block timestamp
+          const contractAddress = ethers.utils.getAddress(event.rawLog.address);
+
+          const parseEvent = (e: keyof typeof ChainEventSigs) =>
+            parseEvmEventToContestEvent(
+              e,
+              contractAddress,
+              timestamp,
+              event.parsedArgs,
+            );
+
+          switch (event.eventSource.eventSignature) {
+            case EvmContestEventSignatures.NewContest:
+              return parseEvent('NewContest') as
+                | {
+                    event_name: EventNames.RecurringContestManagerDeployed;
+                    event_payload: z.infer<
+                      typeof RecurringContestManagerDeployed
+                    >;
+                  }
+                | {
+                    event_name: EventNames.OneOffContestManagerDeployed;
+                    event_payload: z.infer<typeof OneOffContestManagerDeployed>;
+                  };
+            case EvmContestEventSignatures.NewRecurringContestStarted:
+              return parseEvent('NewRecurringContestStarted') as {
+                event_name: EventNames.ContestStarted;
+                event_payload: z.infer<typeof ContestStarted>;
+              };
+            case EvmContestEventSignatures.NewSingleContestStarted:
+              return parseEvent('NewSingleContestStarted') as {
+                event_name: EventNames.ContestStarted;
+                event_payload: z.infer<typeof ContestStarted>;
+              };
+            case EvmContestEventSignatures.ContentAdded:
+              return parseEvent('ContentAdded') as {
+                event_name: EventNames.ContestContentAdded;
+                event_payload: z.infer<typeof ContestContentAdded>;
+              };
+            case EvmContestEventSignatures.VoterVoted:
+              return parseEvent('VoterVoted') as {
+                event_name: EventNames.ContestContentUpvoted;
+                event_payload: z.infer<typeof ContestContentUpvoted>;
+              };
+          }
+
+          return null;
+        })
+        .filter(Boolean);
+
+      await emitEvent(models.Outbox, contestEvents, transaction);
```

Comment:
Doesn't this mean contest events will be emitted twice since there is another `emitEvent` call below that emits all events as well?

Reply:
Yes. Contest events are sent to the contest projection handlers in mapped format, and also to the chain event created policy in raw format. But I suppose since they're not consumed by the chain event created policy, there's no need to have the duplicate data.

Reply:
Yea we should emit in just a single format and break down ChainEventCreated eventually per my comment so we are eventually consistent in our handling.

Reply:
Updated to only emit once